src/Game/Game.elm

```diff
@@ -101,7 +103,9 @@ update msg model =
         Keyboard (KeyDir dir) ->
             let
                 model' =
-                    Game.Collision.tryMoveHero dir model
+                    model
+                        |> Game.Collision.tryMoveHero dir
+                        |> updateViewportOffset model.hero.position
```

Comment:
I *really* like this... and just remembered that I did this in my Room.elm (generate) very nice.